src/emitter.js

```diff
@@ -0,0 +1,279 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * Mixin that injects the events API into its host.
+ *
+ * @class Emitter
+ * @static
+ */
+
+CKEDITOR.define( [ 'eventinfo', 'utils' ], function( EventInfo, utils ) {
+	var EmmitterMixin = {
+		/**
+		 * Registers a callback function to be executed when an event is fired.
+		 *
+		 * @param {String} event The name of the event.
+		 * @param {Function} callback The function to be called on event.
+		 * @param {Object} [ctx] The object that represents `this` in the callback. Defaults to the object firing the
+		 * event.
+		 * @param {Number} [priority] The priority of this callback in relation to other callbacks to that same event.
+		 * Lower values are called first. Defaults to `10`.
```

Comment:
I never liked the default == 10. It's like you have a space 0-9 to do something earlier, and a lot more space after that. But then you end up with -1, -2, etc., because with time it turns out that you need more... I feel that it will be cleaner to have default == 0.